fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/Operator.java

```diff
@@ -56,6 +58,14 @@ public void setPredicate(ScalarOperator predicate) {
         this.predicate = predicate;
     }
 
+    public Projection getProjection() {
+        return projection;
+    }
+
+    public void setProjection(Projection projection) {
```

Comment:
if projection can change, groupExpression may insert into wrong group?

Reply:
Currently setProjection is only used in AddDecodeNodeForDictStringRule, there are two calls, in order to prevent too many changes in a PR. I will delete setProjection in a future PR